gips/00XX-indexing-payments.md

```diff
@@ -0,0 +1,154 @@
+
+### Minimum Viable Product
+
+While this proposal mostly focuses on the full Graph Horizon backed implementation, we are also developing an MVP that does not require smart contracts.
+
+In this MVP, we will roll out a new gateway component (the “dipper”) that will be controlled with a CLI. E&N users (e.g. the BD team) can be onboarded to this CLI and use it to add subgraphs to the list of supported subgraphs. Once a subgraph is in the list, the dipper will take care of finding Indexers and setting up indexing agreements with them.
+
+The Indexer stack will be updated to be able to accept indexing agreements from the dipper when the price is above the price set by the Indexer, and up to a maximum amount of subgraphs also set by the Indexer.
+
+Pricing between Indexer and gateway in the MVP will be based on the "amount of work done" proxy detailed above.
+
+The dipper will initially set up agreements with 3 Indexers for each desired subgraph, as we estimate this is a reasonable number to achieve 99.9% uptime in most subgraphs.
+
+Once the smart contracts proposed in this GIP are deployed, we anticipate adapting the indexer and gateway components accordingly.
+
+This MVP has different trust implications than what this proposal has described so far. In particular these are:
+
+- Indexers do the work upfront and trust that the gateway will pay them accordingly after the fact, since the gateway issues payment vouchers upon receiving POIs and not before.
+- Gateways trust that indexers are reporting the correct amount of work done, since this information is not posted on-chain and thus it's non-slashable. This is partially mitigated by the fact that the gateway can (theoretically) compare the reported work done among the pool of indexers selected and thus can weed out bad actors, but it's nevertheless a weaker mechanism than slashing.
+
+### Generalized collector
+
+[TODO]: # "describe new generalized collector"
+
+## Detailed Specification
+
+_(Optional) Include specific APIs, semantics, data structures, code snippets, etc. related to this proposal. This type of info is required for a proposal to reach the "Draft" stage._
+
+## Backward Compatibility
+
+_(Optional) How does this protocol impact backward compatibility? What breaking changes, if any, are included? Does the proposal have at least N-1 compatibility, or must it be deployed in a knife-edge rollout?_
+
+## Dependencies
+
+(_Optional) How does this proposal depend on other GIPs or other engineering work?_
+
+## Risks and Security Considerations
+
+_(Optional) What technical or security risks are there with implementing this proposal? How might they be mitigated or addressed?_
+
+## Validation
+
+_(Optional) How should this proposal be validated before being included in the protocol? Examples might include security audits, economic modeling, user research, running in testnet, etc._
+
+## Rationale and Alternatives
```

Comment:
It might be worth adding more on why the specific approach for payments is taken, like here: https://github.com/graphprotocol/graph-improvement-proposals/pull/55/files#diff-7596d8f39f5be68b0d93059a0e6f217bcf0828e5c6b7c63be34326df13828bc4R87-R121 - as we already got a lot of questions around that